kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala

```diff
@@ -56,6 +58,24 @@ class KyuubiSyncThriftClient private (
   private var engineAliveThreadPool: ScheduledExecutorService = _
   @volatile private var engineLastAlive: Long = _
 
+  private var asyncRequestExecutor: ScheduledExecutorService = _
```

Comment:
Use something like com.google.common.util.concurrent.MoreExecutors#sameThreadExecutor？

Reply:
seems that we can interrupt the task running in the same thread executor.

Reply:
maybe we don't need the Scheduled interfaces

Reply:
thanks, addressed.